### ISSUE TITLE
FileReference tests use memory based file system

### ISSUE THREAD

**Issue by kurtkilpela** [GemTalk/FileSystemGs#35]
FileReference tests should be run against each FileSystem to ensure high-level behavior is properly tested against each file system. The tests should be factored to use a sandbox directory. Test files and directories should be created in the sandbox.

The sandbox for disk-based file systems should be within an appropriate temporary directory.

**Comment by kurtkilpela** [comment]
FileReferenceTest now provides a sandbox reference for the test cases to use as a root. Each test needs to be evaluated to make it work against the sandbox directory instead of the root of the file system. Several methods have been implemented in DiskStore to support the tests. Many of the outright failing tests have been updated to pass for both a memory-based FileSystem as well as a disk-based FileSystem. In 28654d963dbced128e5d154f6a4a518a2ec40273 there are still several failing/erroring test cases for DiskFileReferenceTest.

**Comment by kurtkilpela** [comment]
The remaining tests have been updated to work against a sandbox directory. Proper assertions were added to several test cases to ensure that more than 'the result was not nil' is tested.

I removed #writeStreamIfPresent: and #writeStreamDo:ifPresent: since they break my expectations. I'd expect the #ifPresent: part to work the same as in #at:ifPresent:. Confusingly, it is treated as an error case here.

See 2ba09c4f512ace5911681802398e44330cbe5dd0
